src/gcp/devConnect.ts

```diff
@@ -181,22 +184,33 @@ export async function listConnections(projectId: string, location: string): Prom
 /**
  * Gets a list of repositories that can be added to the provided Connection.
  */
-export async function fetchLinkableGitRepositories(
+export async function listAllLinkableGitRepositories(
   projectId: string,
   location: string,
   connectionId: string,
-  pageToken = "",
-  pageSize = 1000,
-): Promise<LinkableGitRepositories> {
+): Promise<LinkableGitRepository[]> {
   const name = `projects/${projectId}/locations/${location}/connections/${connectionId}:fetchLinkableRepositories`;
-  const res = await client.get<LinkableGitRepositories>(name, {
-    queryParams: {
-      pageSize,
-      pageToken,
-    },
-  });
+  const repos: LinkableGitRepository[] = [];
 
-  return res.body;
+  const getNextPage = async (pageToken = ""): Promise<void> => {
+    const res = await client.get<LinkableGitRepositories>(name, {
+      queryParams: {
+        PAGE_SIZE_MAX,
```

Comment:
ooc why default to 100 here instead of 1000 if our goal is to list all results?

Reply:
no specific reason. But 1000 makes sense for both I think. Less calls to the APIs. Even if 1000 is more than the supported max for page_size, the API should just return the max page_size it can from what I understand.